Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/features/settings/ui/SettingsFragmentPresenter.kt

```diff
@@ -914,6 +916,85 @@ class SettingsFragmentPresenter(
         )
     }
 
+    private fun addAchievementSettings(sl: ArrayList<SettingsItem>) {
+        val achievementsEnabledSetting: AbstractBooleanSetting = object : AbstractBooleanSetting {
+            override val boolean: Boolean
+                get() = BooleanSetting.ACHIEVEMENTS_ENABLED.boolean
+
+            override fun setBoolean(settings: Settings, newValue: Boolean) {
+                BooleanSetting.ACHIEVEMENTS_ENABLED.setBoolean(settings, newValue)
+                if (newValue)
+                    AchievementModel.init()
+                else
+                    AchievementModel.shutdown()
+                loadSettingsList()
+            }
+
+            override val isOverridden: Boolean
+                get() = BooleanSetting.ACHIEVEMENTS_ENABLED.isOverridden
+
+            override val isRuntimeEditable: Boolean
+                get() = BooleanSetting.ACHIEVEMENTS_ENABLED.isRuntimeEditable
+
+            override fun delete(settings: Settings): Boolean {
+                val result = BooleanSetting.ACHIEVEMENTS_ENABLED.delete(settings)
+                loadSettingsList()
```

Comment:
Again, whatever updates you need to perform in `setBoolean`, you should also perform here.